frontend/packages/console-dynamic-plugin-sdk/src/app/k8s/api-discovery/api-discovery.ts

```diff
@@ -0,0 +1,174 @@
+    .filter(({ name }) => !name.includes('/'))
+    .map(({ name, singularName, namespaced, kind, verbs, shortNames }) => {
+      return {
+        kind,
+        namespaced,
+        verbs,
+        shortNames,
+        label: kind,
+        plural: name,
+        apiVersion,
+        abbr: kindToAbbr(kind),
+        ...(apiGroup ? { apiGroup } : {}),
+        labelPlural: pluralizeKind(kind),
+        path: name,
+        id: singularName,
+        crd: true,
+      };
+    });
+};
+
+const getResources = async (): Promise<DiscoveryResources> => {
+  let apiResourceData;
+  try {
+    apiResourceData = await consoleFetchJSON(`${k8sBasePath}/apis`);
+  } catch (err) {
+    throw err;
+  }
+  const groupVersionMap = apiResourceData.groups.reduce(
+    (acc, { name, versions, preferredVersion: { version } }) => {
+      acc[name] = {
+        versions: _.map(versions, 'version'),
+        preferredVersion: version,
+      };
+      return acc;
+    },
+    {},
+  );
+  const all: Promise<APIResourceList>[] = _.flatten(
+    apiResourceData.groups.map((group) =>
+      group.versions.map((version) => `/apis/${version.groupVersion}`),
+    ),
+  )
+    .concat(['/api/v1'])
+    .map((p) => consoleFetchJSON(`api/kubernetes${p}`).catch((err) => err));
+
+  return Promise.all(all).then((data) => {
+    const resourceSet = new Set<string>();
+    const namespacedSet = new Set<string>();
+    data.forEach(
+      (d) =>
+        d.resources &&
+        d.resources.forEach(({ namespaced, name }) => {
+          resourceSet.add(name);
+          namespaced && namespacedSet.add(name);
+        }),
+    );
+    const allResources = [...resourceSet].sort();
+
+    const safeResources = [];
+    const adminResources = [];
+    const models = _.flatten(data.filter((d) => d.resources).map(defineModels));
+    const coreResources = new Set([
+      'roles',
+      'rolebindings',
+      'clusterroles',
+      'clusterrolebindings',
+      'thirdpartyresources',
+      'nodes',
+      'secrets',
+    ]);
+    allResources.forEach((r) =>
+      coreResources.has(r.split('/')[0]) ? adminResources.push(r) : safeResources.push(r),
+    );
+    const configResources = _.filter(
+      models,
+      (m) => m.apiGroup === 'config.openshift.io' && m.kind !== 'ClusterOperator',
+    );
+    const clusterOperatorConfigResources = _.filter(
+      models,
+      (m) => m.apiGroup === 'operator.openshift.io',
+    );
+
+    return {
+      allResources,
+      safeResources,
+      adminResources,
+      configResources,
+      clusterOperatorConfigResources,
+      namespacedSet,
+      models,
+      groupVersionMap,
+    } as DiscoveryResources;
+  });
+};
+
+const updateResources = () => (dispatch: Dispatch) => {
+  dispatch(getResourcesInFlight());
+
+  getResources()
+    .then((resources) => {
+      // Cache the resources whenever discovery completes to improve console load times.
+      cacheResources(resources);
+      dispatch(receivedResources(resources));
```

Comment:
`receivedResources` is in SDK fo dispatched to that as it updates all resources, can create new selectors and break it down to only have models and this will update only `k8s.resources.models` in store. 

But then testing default API discovery will not be easy as console needs other resources as well to be set for testing the changes